src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -357,9 +371,10 @@ private static void StatusChangedCallback(C4Replicator* repl, C4ReplicatorStatus
             bool transient;
             if ((replicator != null && replicator.IsPermanentError(status.error, out transient)) 
                 && status.level == C4ReplicatorActivityLevel.Stopped) {
-                var array = replicator?._conflictTasks?.Keys?.ToArray();
+                var array = replicator?._conflictTasks?.ToArray();
                 if (array != null) {
                     Task.WaitAll(array);
+                    replicator._conflictTasks = new ConcurrentBag<Task>();
```

Comment:
This new implementation means that this bag will never be cleared unless a replicator stops.  So continuous replication will grow unbounded.